packages/core/src/initialize.ts

```diff
@@ -12,17 +13,26 @@ export function initializeAgent(config: Config): Agent {
   const transports = initializeTransports(config);
   const api = initializeAPI(config, transports, metas);
 
+  const shutdown = () => {
+    shutdownInstrumentations(config);
+    transports.shutdown()
+    if (!agent.config.preventGlobalExposure) {
+      // @ts-ignore
+      delete globalObject[agent.config.globalObjectKey]
+    }
+  }
+
   const agent = initializeGlobalAgent({
     config,
     metas,
     transports,
     api,
+    shutdown,
   });
 
   if (!agent.config.preventGlobalExposure) {
-    Object.defineProperty(globalObject, agent.config.globalObjectKey, {
-      value: agent,
-    });
+    // @ts-ignore
```

Comment:
the problem with `defineProperty` is that it seems it can't be undone. Can't `delete` it or assign to `undefined`,  there's no `undefineProperty`, it's permanent :/ 

- [ ] comment